engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java

```diff
@@ -250,6 +250,24 @@ public void cleanupBackoffsWithoutJobs() {
         });
   }
 
+  @Override
+  public void updateJobDeadline(final long jobKey, final long newDeadline) {
+    this.jobKey.wrapLong(jobKey);
+    final JobRecord job = getJob(jobKey);
+
+    if (job != null) {
+      final long deadline = job.getDeadline();
```

Comment:
🔧 Let's rename `deadline` to `oldDeadline` to keep a clearer distinction between the old and new.

Reply:
Alternatively, I'd also be fine with inlining this variable since we only use it in one place.

```java
deadlineKey.wrapLong(job.getDeadline());
```